mlir/lib/Target/LLVMIR/Dialect/GENX/GENXToLLVMIRTranslation.cpp

```diff
@@ -40,6 +40,44 @@ static llvm::Value *createDeviceFunctionCall(llvm::IRBuilderBase &builder,
   return builder.CreateCall(fn, args);
 }
 
+static llvm::Value *createSubGroupShuffle(llvm::IRBuilderBase &builder,
+                                          llvm::Value *value, llvm::Value *mask,
+                                          GENX::ShflKind kind) {
+  std::string fnName = "";
+  switch (kind) {
+  case GENX::ShflKind::XOR:
+    fnName = "_Z21sub_group_shuffle_xor";
+    break;
+  case GENX::ShflKind::UP:
+    fnName = "_Z20sub_group_shuffle_up";
+    break;
+  case GENX::ShflKind::DOWN:
+    fnName = "_Z22sub_group_shuffle_down";
+    break;
+  };
+  llvm::Type *ty = value->getType();
+  if (ty->isHalfTy())
```

Comment:
Use a TypeSwitch

Reply:
How strongly you want to use TypeSwitch? I also wanted to use TypeSwitch in the beginning, and find that the current way is easier.

Reply:
Ok I see. I am fine with the current code then.